### PR TITLE
[TG-2892] Add builtin function class for string_of_int

### PR DESCRIPTION
**Pull request by romainbrenguier**
This makes the analysis of programs with int to string conversion whose result does not affect the control flow of the program more efficient. Tested on my computer, the `checkDet` test runs in 0.15s instead of 3s and the `checkNonDet` in 0.5s instead of 8s.

### CODE REVIEW

**Review by kroening — APPROVED** [review]

**Review by smowton — COMMENTED** [review]
Should include tests to check what happens when we *do* need the string representation? For example, `String.valueOf(100).charAt(1)` would presumably require us to make and then break a string-of-int object?

**Comment by romainbrenguier** [comment]
@smowton I added a test for the case where the result matters in verification.

I also fixed the case where the radix is different from 10 (we can't use an existing c++ function for that case).

**Review by smowton — APPROVED** [review]

File: src/solvers/refinement/string_builtin_function.cpp [hunk 1]
```diff
@@ -222,6 +222,85 @@ optionalt<exprt> string_insertion_builtin_functiont::eval(
...
+  const std::size_t upper_bound =
+    max_printed_string_length(arg.type(), radix_value);
+  const exprt negative_arg =
+    binary_relation_exprt(arg, ID_le, from_integer(0, type));
+  const exprt absolute_arg =
+    if_exprt(negative_arg, unary_minus_exprt(arg), arg);
+
+  exprt size_expr = from_integer(1, type);
+  std::size_t current_size = 2;
+  for(exprt min_int_with_current_size = radix; current_size <= upper_bound + 1;
```
Comment (peterschrammel) [inline src/solvers/refinement/string_builtin_function.cpp]: To make this loop more readable I would only put `current_size` into`for()`.

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 8de07de91854f7f0ba0640e6aa1fda627e1eee81).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80729163

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 3a1593b1ae07cdbcf24a29085c8a6a00c4363f08).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80735776

**Pull request merged**